doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md

```diff
@@ -150,12 +150,66 @@ The `newClient()` static method receives two mandatory arguments -- the platform
 
 The first argument can be either `DesktopKaaPlatformContext` for Java desktop or `AndroidKaaPlatformContext` for Android applications.
 
-In this example, the second argument is the default implementation of `KaaClientStateListener` -- `SimpleKaaClientStateListener`, that is solely used to log the client state changes.
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of it's constructors. `DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. You can change it by calling corresponding constructor with specific `ExecutorContext` implementation. 
+
+`FlexibleExecutorContext` has four threads group, which responsible for life cycle, callbacks, api and scheduling. Each of them has minimum and maximum threads amount and maximum idle time. Default minimum threads amount for all these groups is zero, maximum threads amount isn't limited and maximum idle time is 100 milliseconds. You can call corresponding constructor for change maximum threads amount or idle time (minimum threads amount is always zero). Except scheduled threads group, you can change only it's minimum threads amount. Recommended to use builder for this purpose. Here some examples for using non-empty constructor and using builder.
+
+<ul class="nav nav-tabs">
+	<li class="active"><a data-toggle="tab" href="#not-empty-constructor">DesktopKaaPlatformContext not empty constructor</a></li>
+	<li><a data-toggle="tab" href="#builder">FlexibleExecutorContext builder</a></li>
+	<li><a data-toggle="tab" href="#simpleExecutorContext">SimpleExecutorContext not empty constructor</a></li>
+</ul>
+
+
+<div class="tab-content">
+<div id="not-empty-constructor" class="tab-pane fade in active" markdown="1">
+
+```java
+KaaClientProperties properties = new KaaClientProperties();
+properties.put(CUSTOM_PROPERTY_NAME, CUSTOM_PROPERTY_VALUE);
+new DesktopKaaPlatformContext(properties,
```

Comment:
it would be useful to add here comment like:
// If you have no properties to set, just pass null instead of first constructor's argument